sdk/src/impl/whirlpool-client-impl.ts

```diff
@@ -91,6 +91,12 @@ export class WhirlpoolClientImpl implements WhirlpoolClient {
     return whirlpools;
   }
 
+  public async getPoolsWithParams(params: GetPoolsParams, refresh = false): Promise<Whirlpool[]> {
+    const pools = await this.ctx.fetcher.listPoolsWithParams(params);
```

Comment:
This method uses `getProgramAccounts`, which always returns the latest whirlpool data. With the `refresh` flag here the API feels weird. Here are two thoughts I have:

1. Add `refresh` to the fetcher and only update the cache for Whirlpools that are not in the cache - even though we have fetched the latest data. This would be used to maintain consistent SDK behavior.
2. Remove `refresh` for `getPoolsWithParams` and have the method always fetch new data

Let me know what you guys think.

Reply:
i'm leaning this method doesn't have a `refresh`. because `refresh = false` doesn't make much sense with the GPA call. 

This method should call this.getPools(addresses, false) though. Otherwise, it's going to make another RPC call to fetch the same data as the GPA call did.